python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -310,43 +316,77 @@ def _stage_and_update_status(self, trials: Iterable[Trial]):
 
         Stages placement groups of all trials.
         """
-        if not self._has_cleaned_up_pgs:
-            # Clean up existing placement groups after trigger the tuning
-            # run step() method for the first time
-            self._pg_manager.cleanup_existing_pg()
-            self._has_cleaned_up_pgs = True
-
         for trial in trials:
+            if len(self._staged_trials) >= self._max_staged_actors:
+                break
+
             if trial.status not in (Trial.PENDING, Trial.PAUSED):
                 continue
+
             if trial in self._staged_trials:
                 continue
-            if self._pg_manager.trial_in_use(trial):
-                continue
 
-            if not self._pg_manager.stage_trial_pg(trial):
-                # Break if we reached the limit of pending placement groups.
-                break
+            resource_request = trial.placement_group_factory
```

Comment:
internally should we unify on `resource_request`?

Reply:
Yes, moving pgf to resource request is part of a separate follow-up PR (https://github.com/ray-project/ray/pull/30028, WIP and not updated with the recent changes)